src/store/names.ts

```diff
@@ -0,0 +1,87 @@
+import { selector } from 'recoil';
+import { getStxAddress } from '@stacks/wallet-sdk';
+
+import { fetcher } from '@common/wrapped-fetch';
+
+import { accountsStore } from '@store/wallet';
+import { currentNetworkStore, currentTransactionVersion } from '@store/networks';
+
+async function fetchNamesByAddress(networkUrl: string, address: string): Promise<string[]> {
+  const res = await fetcher(networkUrl + `/v1/addresses/stacks/${address}`);
+  const data = await res.json();
+  return data?.names || [];
+}
+
+function makeKey(networkUrl: string, address: string): string {
+  return `${networkUrl}__${address}`;
+}
+
+function getLocalNames(networkUrl: string, address: string): string[] | null {
+  const key = makeKey(networkUrl, address);
+  const value = localStorage.getItem(key);
+  return value ? JSON.parse(value) : null;
+}
+
+function setLocalNames(networkUrl: string, address: string, names: string[]): void {
+  const key = makeKey(networkUrl, address);
+  return localStorage.setItem(key, JSON.stringify(names));
+}
+
+type AccountNameState =
+  | {
+      address: string;
+      index: number;
+      names: string[];
+    }[]
+  | null;
```

Comment:
```suggestion
interface AccountName {
  address: string;
  index: number;
  names: string[];
}
type AccountNameState =  AccountName[] | null;
```
Makes type reuse easier